src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngineResult.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public abstract class RazorProjectEngineResult
+    {
+        public abstract RazorCodeDocument CodeDocument { get; }
+
```

Comment:
Today this is consumed in two ways:
- Grabbing the CodeDocument and passing it around tooling
- Grabbing the CSharp document and spitting it out somewhere.

Should we add convenience properties or methods around the diagnostics (success) of the result? Or the csharp document?